src/dom/Integer.ts

```diff
@@ -100,13 +100,13 @@ export class Integer extends Value(Number, IonTypes.INT, _fromJsConstructor) {
       // `compareOnlyIon` requires that the provided value be an ion.dom.Integer instance.
       if (other instanceof Integer) {
         isSupportedType = true;
-        valueToCompare = other.numberValue();
+        valueToCompare = other.bigIntValue();
```

Comment:
I think this means that two `ion.dom.Integer` instances representing small numbers won't be able to compare their `number` values without first converting them to `JSBI`. Is that right?

I think you could side step that extra construction by testing whether `valueToCompare` is a `JSBI` or not right before the final `return` and doing the appropriate comparison.

```typescript
if (other instanceof Integer) {
  isSupportedType = true;
  if (this.bigIntValue == null && other.bigIntValue == null) {
    valueToCompare = other.numberValue();
  } else {
    // One of them is a JSBI
    valueToCompare = other.bigIntValue();
  }
}
```

```typescript
if (other instanceof global.Number || typeof other === "number") {
  isSupportedType = true;
  if (this.bigIntValue == null) {
    valueToCompare = other.valueOf();
  } else {
    valueToCompare = JSBI.BigInt(other.valueOf());
  }
}
```

```typescript
if (valueToCompare instanceof JSBI) {
  return  JSBI.equal(this.bigIntValue(), valueToCompare);
}
return this.numberValue() == valueToCompare;
```